lxd/oidc_sessions.go

```diff
@@ -0,0 +1,313 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/lxd/lxd/auth"
+	"github.com/canonical/lxd/lxd/db"
+	dbCluster "github.com/canonical/lxd/lxd/db/cluster"
+	"github.com/canonical/lxd/lxd/request"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/entity"
+	"github.com/canonical/lxd/shared/version"
+)
+
+var oidcSessionsCmd = APIEndpoint{
+	Name:        "oidc_sessions",
+	Path:        "auth/oidc-sessions",
+	MetricsType: entity.TypeIdentity,
+	Get: APIEndpointAction{
+		Handler:       oidcSessionsGet,
+		AccessHandler: allowAuthenticated,
+	},
+}
+
+var oidcSessionCmd = APIEndpoint{
+	Name:        "oidc_session",
+	Path:        "auth/oidc-sessions/{id}",
+	MetricsType: entity.TypeIdentity,
+	Get: APIEndpointAction{
+		Handler:       oidcSessionGet,
+		AccessHandler: oidcSessionAccessHandler(auth.EntitlementCanView),
+	},
+	Delete: APIEndpointAction{
+		Handler:       oidcSessionDelete,
+		AccessHandler: oidcSessionAccessHandler(auth.EntitlementCanDelete),
```

Comment:
This is `can_delete` so that oidc identities can delete their own sessions without any special logic.